javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,433 @@
+        public int hashCode() {
+            return getDelegate().hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return getDelegate().mkString("[", ", ", "]");
+        }
+
+        // -- private helpers
+
+        private boolean setDelegateAndCheckChanged(Seq<T> delegate) {
+            return setDelegate(delegate) != delegate;
+        }
+
+        private T setDelegateAndGetPreviousElement(int index, Seq<T> delegate) {
+            return setDelegate(delegate).get(index);
+        }
+
+        // DEV-Note: An iterator is intentionally not serializable.
+        private static class ListIterator<T, C extends Seq<T>> implements java.util.ListIterator<T> {
+
+            private ListView<T, C> list;
+            private int index;
+            private boolean dirty = true;
+
+            ListIterator(ListView<T, C> list, int index) {
+                this.list = list;
+                this.index = index;
+            }
+
+            @Override
+            public boolean hasNext() {
+                return index < list.size();
+            }
+
+            @Override
+            public T next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                dirty = false;
+                return list.get(index++);
+            }
+
+            @Override
+            public int nextIndex() {
+                return index;
+            }
+
+            @Override
+            public boolean hasPrevious() {
+                return index > 0;
+            }
+
+            @Override
+            public T previous() {
+                if (!hasPrevious()) {
+                    throw new NoSuchElementException();
+                }
+                dirty = false;
+                return list.get(--index);
+            }
+
+            @Override
+            public int previousIndex() {
+                return index - 1;
+            }
+
+            @Override
+            public void remove() {
+                checkDirty();
+                // may throw a ClassCastException accordingly to j.u.ListIterator.remove()
+                list.remove(index);
+                dirty = true;
+            }
+
+            @Override
+            public void set(T value) {
+                checkDirty();
+                // may throw a ClassCastException accordingly to j.u.ListIterator.set(T)
+                list.set(index, value);
+            }
+
+            @Override
+            public void add(T value) {
+                /* TODO:
+                 * The new element is inserted before the implicit
+                 * cursor: a subsequent call to {@code next} would be unaffected, and a
+                 * subsequent call to {@code previous} would return the new element.
+                 * (This call increases by one the value that would be returned by a
+                 * call to {@code nextIndex} or {@code previousIndex}.)
+                 */
```

Comment:
we will need good tests for set, add and remove in conjunction with next/prev and hasNext/hasPrev. This is tricky stuff when the ListView is mutable...